### PR TITLE
WIP Merge upstream

### PR DESCRIPTION
**Pull request by justinuang**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by robert3005** [comment]
You have conflicts in custom pod spec code since we merged the pr before it merged upstream - would be good to just take the upstream version

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -191,24 +191,6 @@ To use a secret through an environment variable use the following options to the
 --conf spark.kubernetes.executor.secretKeyRef.ENV_NAME=name:key
 ```
 
-<<<<<<< HEAD
```
Comment (justinuang) [inline docs/running-on-kubernetes.md]: reverted to upstream.

File: pom.xml [hunk 2]
```diff
@@ -130,21 +130,14 @@
...
-    <orc.version>1.5.2</orc.version>
-||||||| merged common ancestors
-    <parquet.version>1.10.0</parquet.version>
-    <orc.version>1.5.2</orc.version>
-=======
-    <parquet.version>1.10.0</parquet.version>
     <orc.version>1.5.3</orc.version>
->>>>>>> upstream/master
     <orc.classifier>nohive</orc.classifier>
+    <hive.parquet.version>1.6.0</hive.parquet.version>
```
Comment (justinuang) [inline pom.xml]: TODO investigate why it disappeared.

File: pom.xml [hunk 3]
```diff
@@ -2196,14 +2129,6 @@
         </exclusions>
       </dependency>
       <dependency>
-=======
-        <groupId>com.twitter</groupId>
```
Comment (justinuang) [inline pom.xml]: @robert3005 not sure why palantir/spark doesn't have  parquet-hadoop-bundle in the merge conflict. your commit removes it c663aa6777, should i just put it back like i did here?

File: resource-managers/kubernetes/docker/src/main/dockerfiles/spark/Dockerfile [hunk 4]
```diff
@@ -31,13 +31,7 @@ ARG k8s_tests=kubernetes/tests
...
 RUN set -ex && \
     apk upgrade --no-cache && \
-<<<<<<< HEAD
-    apk add --no-cache bash tini && \
-||||||| merged common ancestors
-    apk add --no-cache bash tini libc6-compat linux-pam && \
-=======
-    apk add --no-cache bash tini libc6-compat linux-pam krb5 krb5-libs && \
->>>>>>> upstream/master
+    apk add --no-cache bash tini krb5 krb5-libs && \
```
[the remark this answers is not in the document]
Comment (justinuang) [reply resource-managers/kubernetes/docker/src/main/dockerfiles/spark/Dockerfile]: Talked offline. Our base image already has it.

**Comment by justinuang** [comment]
Closing in favor of https://github.com/palantir/spark/pull/431

**Pull request closed without merging**